src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs

```diff
@@ -126,7 +126,7 @@ public ActionResult IndexFilterPOST(TemplateIndexViewModel model)
                 { _optionsSearch, model.Options.Search }
             });
 
-        public async Task<IActionResult> Create(bool adminTemplates = false, string returnUrl = null)
+        public async Task<IActionResult> Create(string name = null, bool adminTemplates = false, string returnUrl = null)
```

Comment:
@microposmp which view in OC is providing the value to the new parameter?

Reply:
There is a RedirectToAction on line 235 (line number after changes in this PR) in the Edit action in the same controller if there is no existing template document available with the name parameter provided.

It's also called from the Index view (list of templates) without the name parameter.